### DISCUSSION TITLE
[Feature] Method for inserting content relative to line number

### DISCUSSION THREAD

**Discussion by bytewife** [coddingtonbear/obsidian-local-rest-api#29] [Ideas]
Hi! This plugin is fantastic- thank you for it! What are your thoughts on adding a method similar to PATCH that instead inserts before or after a line number? I believe it would be more flexible & conflict-free than the headers-based approach. My use case is inserting before/after bullet points; I prefer an outlining approach to notes, rather than a header-based approach.

**Comment by coddingtonbear** [comment]
It seems like inserting relative to a line number would be something you can pull off easily by just:

1. `GET`-ing the content of the relevant file.
2. Inserting the content at the line number you want.
3. `PUT`-ing a the file content again.

The operations there are so simple that I'm not sure paving it over with an API endpoint adds all that much value relative to the amount of complexity it'd add; so I'm not sure I'm all that motivated to add/maintain that.

Something I _am_ considering adding support for is inserting content relative to block references (see #26), though, and that might help you with your use case?
